### PR TITLE
Multilng client generator

### PR DESCRIPTION
**Pull request by slemrmartin**
rake task for generating client in more languages, defaulting ruby

### CODE REVIEW

**Comment by slemrmartin** [comment]
@abellotti please review

File: lib/tasks/client_generate.rake [hunk 1]
```diff
@@ -61,15 +61,22 @@ class ClientGenerator
...
-  desc "Generate the Toplogical Inventory API Ruby Client"
+  desc "Generate the Topological Inventory API Ruby Client"
   task :generate, [:client_dir] => [:environment] do |_task, args|
     default_client_dir = Pathname.new(Rails.root.join("..", "topological_inventory-api-client-ruby"))
     args.with_defaults(:client_dir => default_client_dir)
-    ClientGenerator.new.generate_ruby_client(args[:client_dir])
+    ClientGenerator.new.generate_client(args[:client_dir], "ruby")
+  end
+
+  desc "Generate the Topological Inventory API Bash Client"
```
Comment (abellotti) [inline lib/tasks/client_generate.rake]: I'd personally just use curl, but yea I think this is a great idea. Can you see if we can use the same rake task just add an optional lang argument defaulting to ruby.  This way we can simply use the same rake task for any other language without having to add a rake task for each ?  Thanks.

Comment (slemrmartin) [reply lib/tasks/client_generate.rake]: @abellotti now it defaults to ruby, but second param can be added as language

Comment (abellotti) [reply lib/tasks/client_generate.rake]: Nice update!! Thanks @slemrmartin 

Can you just update the usage in the comment section ? that'll great.
#        bundle exec rake client:generate[/alternate_ruby_client_dir]

Thanks.

Comment (slemrmartin) [reply lib/tasks/client_generate.rake]: yes, and I'll create similar pr to sources-api too.

File: lib/tasks/client_generate.rake [hunk 2]
```diff
@@ -61,15 +61,15 @@ class ClientGenerator
...
+    system("java -jar #{generator_cli_jar} generate -i #{openapi_yaml_file} -c #{generator_config} -g #{lng} -o #{client_dir}")
   end
 end
 
 namespace :client do
-  desc "Generate the Toplogical Inventory API Ruby Client"
-  task :generate, [:client_dir] => [:environment] do |_task, args|
+  desc "Generate the Topological Inventory API Ruby Client (default is Ruby)"
+  task :generate, [:client_dir, :language] => [:environment] do |_task, args|
     default_client_dir = Pathname.new(Rails.root.join("..", "topological_inventory-api-client-ruby"))
```
Comment (abellotti) [inline lib/tasks/client_generate.rake]: hmm, this should probably change  to -api-client-#{language}

File: lib/tasks/client_generate.rake (hunk 1 above)
Comment (slemrmartin) [reply lib/tasks/client_generate.rake]: done

File: lib/tasks/client_generate.rake (hunk 2 above)
Comment (slemrmartin) [reply lib/tasks/client_generate.rake]: done

**Comment by abellotti** [comment]
Thanks @slemrmartin for the update 👍 

/cc @agrare can you take a 👀

**Comment by agrare** [comment]
Love the flexibility, is there anything we should add to the openapi config json slash do we need different config files per language?

**Comment by slemrmartin** [comment]
@agrare I tried only bash client, it works correctly generated with the same json config

**Comment by agrare** [comment]
Okay, something we should keep in mind if we go to generate e.g. python/go clients...they might need different json config.

**Pull request merged**